packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -505,7 +505,7 @@ class AppDomain extends Domain {
   }
 }
 
-typedef void _DeviceEventHandler(Device device);
+typedef Future<Null> _DeviceEventHandler(Device device);
```

Comment:
`Stream.listen()` doesn't support async callbacks - it won't await the listener when notifying it.